api/src/main/java/org/openmrs/module/registrationcore/api/mpi/common/MpiAuthenticator.java

```diff
@@ -0,0 +1,28 @@
+package org.openmrs.module.registrationcore.api.mpi.common;
+
+/**
+ * This service perform authentication to MPI server. It is a Spring managed bean which is configured
+ * in moduleApplicationContext.xml.
+ */
+public interface MpiAuthenticator {
```

Comment:
I wonder if you have prematurely created an interface for this. Are we sure that this will actually be sufficient for other MPI tools besides OpenEMPI?